pkg/util/portforward/exec.go

```diff
@@ -0,0 +1,52 @@
+package portforward
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+// ExecStdout executes a command in the given namespace/pod/container and
+// streams its stdout.
+func ExecStdout(ctx context.Context, log *logrus.Entry, restconfig *rest.Config, namespace, pod, container string, command []string) (io.ReadCloser, error) {
+	v := url.Values{
+		"container": []string{container},
+		"command":   command,
+		"stdout":    []string{"true"},
+	}
+
+	spdyConn, err := dialSpdy(ctx, restconfig, "/api/v1/namespaces/"+namespace+"/pods/"+pod+"/exec?"+v.Encode())
```

Comment:
This part of the code gives an `unexpected http status code 403` during the exec step of the action.